plugins/circleci/src/index.ts

```diff
@@ -1,16 +1,18 @@
 import { writeState } from '@dotcom-tool-kit/state'
-import CircleCiConfigHook from './circleci-config'
+import CircleCiConfigHook, { generateConfigWithJob } from './circleci-config'
 
 export class BuildCI extends CircleCiConfigHook {
   job = 'tool-kit/build'
-  jobOptions = { requires: ['tool-kit/setup'] }
-  addToNightly = true
+  config = generateConfigWithJob({ name: this.job, requires: ['tool-kit/setup'], addToNightly: true })
 }
 
 export class TestCI extends CircleCiConfigHook {
   job = 'tool-kit/test'
-  jobOptions = { requires: [new BuildCI(this.logger).job] }
-  addToNightly = true
+  config = generateConfigWithJob({
+    name: this.job,
+    requires: [new BuildCI(this.logger).job],
```

Comment:
is `this.logger` the logger defined in the abstract `Hook` class [here](https://github.com/Financial-Times/dotcom-tool-kit/blob/main/lib/types/src/index.ts#L173)? 

If so, (and even if not so 😁 ) why does it need to be passed into `new BuildCI`?

Reply:
yes it is, and it needs to be passed because it's specified as an argument to the constructor [here](https://github.com/Financial-Times/dotcom-tool-kit/blob/main/lib/types/src/index.ts#L189)

ALTHOUGH, I realised during the demo this morning, and have been reminded now, that now that the `job` isn't referenced anywhere in the abstract class we can make it a `static` field which means other classes can reference a class's job without instantiating an instance. new commit incoming...